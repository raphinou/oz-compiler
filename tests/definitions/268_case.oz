% Pattern conjunction
local
   R=rec(inner(a b))

   A=a
   B=b
in
   case R
   of rec(Foo=inner(Capt b)) then
      {Show Foo}
      {Show Capt}
   [] rec(!A !B c) then
      {Show error}
   [] rec( b c) then
      {Show error}
   else
      {Show error}
   end

   case R
   of rec(inner(Capt !B)=Foo) then
      {Show Foo}
      {Show Capt}
   [] rec(!A !B c) then
      {Show error}
   [] rec( b c) then
      {Show error}
   else
      {Show error}
   end

   case R
   of rec(inner(a Capt)) then
      {Show Capt}
   [] rec(!A !B c) then
      {Show error}
   [] rec( b c) then
      {Show error}
   else
      {Show error}
   end
end

