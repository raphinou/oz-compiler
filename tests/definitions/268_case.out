inner(a b)
a
inner(a b)
a
b
